### PR TITLE
crimson/os/seastore/cache: cleanups and comments

### PR DESCRIPTION
**Pull request by cyx1231st**
I'm looking at the boundary of Cache component.

This might be a good basis to proceed to fix cache metrics and fine-grained-cache (https://github.com/ceph/ceph/pull/57787)

Manual rough tests looks good. @xxhdx1985126 Does the revised version of `get_extent_viewable_by_trans()` look reasonable to you?

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by xxhdx1985126 — APPROVED** [review]
LGTM!

**Comment by cyx1231st** [comment]
Change-set: removed an useless assert.

**Comment by cyx1231st** [comment]
Change-set: noticed a duplicated call to `touch_extent()` and removed.

**Comment by cyx1231st** [comment]
jenkins retest this please

File: src/crimson/os/seastore/cache.h [hunk 1]
```diff
@@ -471,18 +476,31 @@ class Cache {
     CachedExtentRef extent)
   {
     assert(extent->is_valid());
+    if (extent->is_pending() && !extent->is_stable_writting()) {
+      assert(extent->is_pending_in_trans(t.get_trans_id()));
+      assert(extent->is_fully_loaded());
```
Comment (cyx1231st) [inline src/crimson/os/seastore/cache.h:481]: Somehow this is violating the UT.

Comment (cyx1231st) [reply src/crimson/os/seastore/cache.h:481]: I see, clean-existing should be read out upon the access.

**Comment by cyx1231st** [comment]
jenkins retest this please

**Comment by cyx1231st** [comment]
changeset:
* rework `get_extent_viewable_by_trans()` to pass the tests: https://github.com/ceph/ceph/pull/58983/commits/6933f4c9980a9db2025f625d64ee4d81868e2728
* introduce `maybe_add_to_read_set()` for `get_extent_viewable_by_trans()` as a special case: https://github.com/ceph/ceph/pull/58983/commits/8c6cedea0af0c6aa10ac09778c9113aff85ada86

cc @xxhdx1985126

**Comment by cyx1231st** [comment]
jenkins test api

**Comment by cyx1231st** [comment]
https://shaman.ceph.com/builds/ceph/wip-yingxin-testing-2024-08-07-1253/
https://pulpito.ceph.com/yingxin-2024-08-07_07:12:11-crimson-rados-wip-yingxin-testing-2024-08-07-1253-distro-default-smithi/
* 7841889/90: https://tracker.ceph.com/issues/63844
* 7841905/06: https://tracker.ceph.com/issues/66761

**Pull request merged**